site/src/routes/repl/create.json.js

```diff
@@ -8,21 +8,14 @@ export async function post(req, res) {
 	if (!user) return; // response already sent
 
 	try {
-		const { name, components } = await body(req);
-
-		const files = {};
-		components.forEach(component => {
-			const text = component.source.trim();
-			if (!text.length) return; // skip empty file
-			files[`${component.name}.${component.type}`] = text;
-		});
+		const { name, files } = await body(req);
 
 		const [row] = await query(`
 			insert into gists(user_id, name, files)
-			values ($1, $2, $3) returning *`, [user.id, name, files]);
+			values ($1, $2, $3) returning *`, [user.id, name, JSON.stringify(files)]);
```

Comment:
Same as above, the `pg` layer should be morphing stuff in a friendly way for the `json` column